events/events_network.c

```diff
@@ -136,6 +136,15 @@ events_network_register(int (*func)(void *), void * cookie, int s, int op)
 		goto err0;
 
 	/*
+	 * There is a new socket/operation pair:
+	 * Rebuild the fd list on the next call to events_network_select().
+	 */
+	if (fds != NULL) {
```

Comment:
I think this problem existed before as well: The fd sets were cleared for every call to select, no? If not I'd like to understand the difference, as I said: I never worked with event based C networking before.

see: https://github.com/Tarsnap/libcperciva/pull/18/files#diff-6745fd3f7945293c64b6fbcc79828177L243